PROTOCOL.md

```diff
@@ -424,6 +429,44 @@ enough to get you started. You can watch the data sent and received from
 the server on a regular connection, or look at the client source code
 for a full list of message types.
 
+**Action requests**
+
+These are how the client sends the player's decisions to the server. Most
+requests are sent with `REQUESTID`. `REQUESTID` is used to identify which
+action the request was intended for to protect from cases of trying to send
+multiple requests for the same action, which would otherwise effect future
+decisions. The first decision of a match has a `REQUESTID` of 1, and each
+successful decision increments the counter by 1.
+
+`|/team ORDER|REQUESTID`
+
+> Chooses the team order. Numbers not listed will be displaced to the back of the list.
+> For example `/team 25` sets the default team order to 251346.
+
+`|/choose move NUMBER TARGET|REQUESTID` or `|/choose switch NUMBER|REQUESTID`
+
+> Requests to use the selected move `NUMBER` on `TARGET` Pokemon or to switch to
+> `NUMBER` Pokemon where `NUMBER` is the position in the team order. `NUMBER` starts
+> at 1 and can be up to 6 for `switch` and 12 (though usually 4) for `move`.
+> Pokemon is ko'd `/choose switch` is used to determine their replacement.
+> In singles `/move NUMBER` and `/switch NUMBER` are used as shorthand, and `TARGET`
+> does not need to be specified.
+> For double and triple battles, decisions are sent for all team positions in the same
+> line separated by commas. Moves which specify a target list their chosen target as a
+> number wherein the opposing Pokemon are positive integers counting up from `1` starting
+> on the right. Ally Pokemon targets are negative integers counting down from `-1`
+> starting on the left. Moves which don't need to specify a target send only the action.
+> If not all positions are making a decision, `pass` is sent in place of the action for
+> the position to be skipped.
+> In triples `/choose shift` requests to `|swap|` to the middle.
+> For example `|/choice move 1 -3,pass,move 4|55` will have the leftmost Pokemon attack
+> it's rightmost ally with it's first move, the middle Pokemon will do nothing, and the
```

Comment:
`pass` also doesn't work like that. You can't just choose to do nothing. If you send `pass` as the choice for a Pokémon, the validator will declare it invalid, and replace it with a valid choice (generally "use the first non-disabled move").

`pass` is only used for fainted Pokémon in doubles/triples, e.g. if you only have two unfainted Pokémon in triples. The fainted Pokémon can't do anything so you send `pass`.